shared/src/main/kotlin/com/egm/stellio/shared/model/CompactedEntity.kt

```diff
@@ -140,27 +158,41 @@ private fun filterLanguageProperty(value: Map<String, Any>, transformationParame
     val attributeCompactedType = value[JSONLD_TYPE_TERM]?.let {
         AttributeCompactedType.forKey(value[JSONLD_TYPE_TERM] as String)
     }
-    val languageFilteredValue =
-        if (attributeCompactedType == LANGUAGEPROPERTY) {
-            val localeRanges = Locale.LanguageRange.parse(languageFilter)
-            val propertyLocales = (value[JSONLD_LANGUAGEMAP_TERM] as Map<String, Any>).keys.sorted()
-            val bestLocaleMatch = Locale.filterTags(localeRanges, propertyLocales)
-                .getOrElse(0) { _ ->
-                    // as the list is sorted, @none is the first in the list if it exists
-                    propertyLocales.first()
-                }
-            mapOf(
-                JSONLD_TYPE_TERM to NGSILD_PROPERTY_TERM,
-                JSONLD_VALUE_TERM to (value[JSONLD_LANGUAGEMAP_TERM] as Map<String, Any>)[bestLocaleMatch],
-                NGSILD_LANG_TERM to bestLocaleMatch
-            )
-        } else value
 
-    return languageFilteredValue.mapValues { entry ->
-        if (entry.key == NGSILD_ENTITY_TERM)
-            (entry.value as CompactedEntity).toFilteredLanguageProperties(languageFilter)
-        else entry.value
-    }
+    if (attributeCompactedType == LANGUAGEPROPERTY) {
+        val localeRanges = Locale.LanguageRange.parse(languageFilter)
+        val propertyLocales = (value[JSONLD_LANGUAGEMAP_TERM] as Map<String, Any>).keys.sorted()
+        val bestLocaleMatch = Locale.filterTags(localeRanges, propertyLocales)
+            .getOrElse(0) { _ ->
+                // as the list is sorted, @none is the first in the list if it exists
+                propertyLocales.first()
+            }
+        val filteredMainAttribute = mutableMapOf(
+            JSONLD_TYPE_TERM to NGSILD_PROPERTY_TERM,
+            JSONLD_VALUE_TERM to (value[JSONLD_LANGUAGEMAP_TERM] as Map<String, Any>)[bestLocaleMatch],
+            NGSILD_LANG_TERM to bestLocaleMatch
+        )
+
+        value.filter { entry ->
+            !JSONLD_COMPACTED_ATTRIBUTE_CORE_MEMBERS.contains(entry.key) && entry.key != JSONLD_LANGUAGEMAP_TERM
```

Comment:
you can replace entry
```suggestion
        value.filter { (key,_) ->
            !JSONLD_COMPACTED_ATTRIBUTE_CORE_MEMBERS.contains(key) && key != JSONLD_LANGUAGEMAP_TERM
```